src/main/scala/io/iohk/ethereum/vm/Blake2bCompression.scala

```diff
@@ -0,0 +1,151 @@
+package io.iohk.ethereum.vm
+
+import java.util.Arrays.copyOfRange
+
+// scalastyle:off magic.number
+object Blake2bCompression {
+  val MessageBytesLength = 213
+
+  import org.bouncycastle.util.Pack
+
+  private val IV: Array[Long] = Array(
+    0x6a09e667f3bcc908L,
+    0xbb67ae8584caa73bL,
+    0x3c6ef372fe94f82bL,
+    0xa54ff53a5f1d36f1L,
+    0x510e527fade682d1L,
+    0x9b05688c2b3e6c1fL,
+    0x1f83d9abfb41bd6bL,
+    0x5be0cd19137e2179L)
+
+  private val PRECOMPUTED: Array[Array[Byte]] = Array(
+    Array(0, 2, 4, 6, 1, 3, 5, 7, 8, 10, 12, 14, 9, 11, 13, 15),
+    Array(14, 4, 9, 13, 10, 8, 15, 6, 1, 0, 11, 5, 12, 2, 7, 3),
+    Array(11, 12, 5, 15, 8, 0, 2, 13, 10, 3, 7, 9, 14, 6, 1, 4),
+    Array(7, 3, 13, 11, 9, 1, 12, 14, 2, 5, 4, 15, 6, 10, 0, 8),
+    Array(9, 5, 2, 10, 0, 7, 4, 15, 14, 11, 6, 3, 1, 12, 8, 13),
+    Array(2, 6, 0, 8, 12, 10, 11, 3, 4, 7, 15, 1, 13, 5, 14, 9),
+    Array(12, 1, 14, 4, 5, 15, 13, 10, 0, 6, 9, 8, 7, 3, 2, 11),
+    Array(13, 7, 12, 3, 11, 14, 1, 9, 5, 15, 8, 2, 0, 4, 6, 10),
+    Array(6, 14, 11, 0, 15, 9, 3, 8, 12, 13, 1, 10, 2, 7, 4, 5),
+    Array(10, 8, 7, 1, 2, 4, 6, 5, 15, 9, 3, 13, 11, 14, 12, 0))
+
+  private def bytesToInt(bytes: Array[Byte]) = Pack.bigEndianToInt(bytes, 0)
+
+  private def bytesToLong(bytes: Array[Byte]) = Pack.littleEndianToLong(bytes, 0)
+
+  def isValidInput(input: Array[Byte]): Boolean= {
+    if (input.length != MessageBytesLength || (input(212) & 0xFE) != 0) {
```

Comment:
Why not just `def isValidInput(input: Array[Byte]): Boolean = !(input.length != MessageBytesLength || (input(212) & 0xFE) != 0)` ?